### DISCUSSION TITLE
podcasts: import to beets, read info from json, import cover from jpg

### DISCUSSION THREAD

**Discussion by prankousky** [beetbox/beets#4655] [Q&A]
Hi everybody,

previously, I exclusively used beets for music and audiobooks. This always worked great. Audiobooks usually needed to be tagged manually before important, but that was no big issue.

Now I am trying to build a perfectly tagged podcast library with beets, but don't know if  this can be automated.

# Scenario

I have

* `<date-title>.mp3`, for example `20190123-Können Namen peinlich sein_.mp3`
* `<date-title>.meta.json`, for example `20190123-Können Namen peinlich sein_.meta.json`
* `<date-title>.jpg`, for example `20190123-Können Namen peinlich sein_.jpg`

In this case, the `(...)sein_.jpg` underscore is only there because the question mark in the title got renamed to an underscore; there won't be a `_` in front of the file extension if the title just included letters and numbers.

**mp3** is obviously the audio file itself.

**json** includes the following (contents of the actual file)

```json
{
    "title": "Können Namen peinlich sein?",
    "contentSnippet": "Sabine Leutheusser-Schnarrenberger und Cheyenne Savannah Ochsenknecht, so heißen Menschen wirklich. Allerdings heißt Dietmar Wischmeyer eben Dietmar Wischmeyer. Er und seine Podcast-Partnerin Tina Voss wundern sich darüber – in Wischmeyers Stundenhotel von Bremen Zwei. Ihr Name geht schon eleganter über die Lippen.",
    "pubDate": "Wed, 23 Jan 2019 21:05:00 +0100"
}
```
and **jpg** is the cover file.

# What I'd like to do

When beets imports the file(s), I'd need it to automatically do the following for each mp3 file using the appropriate json and jpg files (which it could identify by their name)

1. set the `title` in metadata to the `title` part from the json file 
2. set the `comment` in metadata to the `contentSnippet` part from the json file
3. set the coverart to the jpg file that is named 
4. either
  4.1 *if possible* set `year` in metadata to the year value from `pubDate` from the json file
  4.2 set `copyright` to `pubDate` from the json file
  4.3 or just do both

Is this possible? If so, I'd only have to manually change the `artist` and perhaps `album` metadata, instead of doing everything by hand.

I use [podcast-dl](https://github.com/lightpohl/podcast-dl) to download my podcasts. Unfortunately, even though I  thought it would, it doesn't create any metadata (tried the different command line options), nor does it write the artwork the audio file. It will download the metadata as json and artwork as jpg, but the mp3 file looks like this

```bash
➜ id3info 20190123-Können\ Namen\ peinlich\ sein_.mp3

*** Tag information for 20190123-Können Namen peinlich sein_.mp3
*** mp3 info
MPEG1/layer III
Bitrate: 128KBps
Frequency: 48KHz

```
There is nothing.

Is there a way to use beets to do this, then import it to my collection?

Thank you in advance for your help :)

**Comment by sampsyo** [comment]
We don't yet have any good way to read import matches from external files, like JSON. That sounds like an interesting plugin to write, however, and doesn't seem like it would be too hard! It could behave like any other "data source" plugin (like the ones for Beatport, Discogs, etc.) but actually just search within a JSON file. Feel free to give it a shot if you're interested!

**Comment by prankousky** [comment]
Thank you @sampsyo 

Unfortunately, my skills aren't adequate; even if I were able to hack something together, I doubt it would be good enough to work for anybody else - that being said, I wouldn't even know were to begin at all, unfortunately.
